File: Software Technologies/Java Basic Syntax Exercises/src/PhonebookUpgrade.java
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.TreeMap;

public class PhonebookUpgrade {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input;

        TreeMap<String, String> phonebook = new TreeMap<>();

        while (!"END".equals(input = scanner.nextLine())) {

            String[] tokens = input.split(" ");

            String command = tokens[0];

            switch (command) {
                case "A":
                    String name = tokens[1];
                    String number = tokens[2];

                    phonebook.put(name, number);
                    break;
                case "S":
                    name = tokens[1];
                    if (phonebook.containsKey(name)) {
                        String foundNumber = phonebook.get(name);
                        System.out.printf("%s -> %s%n",name, foundNumber );
                    } else {
                        System.out.printf("Contact %s does not exist. %n", name);
                    }
                    break;
                case "ListAll":
                    for (Map.Entry<String, String> entry : phonebook.entrySet()) {
                        System.out.printf("%s -> %s%n", entry.getKey(), entry.getValue());
                    }
                    break;
            }
        }
    }
}

File: Programming Fundamentals/Programming-Fundamentals-September-2017/DictionariesLambdaAndLINQ-Exercise/02.PhonebookUpgrade/PhonebookUpgrade.cs
namespace _02.PhonebookUpgrade
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    public class Phonebook
    {
        public static void Main()
        {
            var phonebook = new SortedDictionary<string, string>();

            string input = Console.ReadLine();

            while (input != "END")
            {
                var commandArguments = input.Split(' ').ToList();

                string command = commandArguments[0];

                if (command == "A")
                {
                    string name = commandArguments[1];
                    string telephone = commandArguments[2];

                    if (!phonebook.ContainsKey(name))
                    {
                        phonebook.Add(name, telephone);
                    }
                    else
                    {
                        phonebook[name] = telephone;
                    }
                }
                else if (command == "S")
                {
                    string name = commandArguments[1];

                    if (phonebook.ContainsKey(name))
                    {
                        Console.WriteLine($"{name} -> {phonebook[name]}");
                    }
                    else
                    {
                        Console.WriteLine($"Contact {name} does not exist.");
                    }
                } else if (command == "ListAll")
                {
                    foreach (var item in phonebook)
                    {
                        Console.WriteLine("{0} -> {1}", item.Key, item.Value);
                    }
                }
                input = Console.ReadLine();
            }
        }
    }
}
